src/command/mod.rs

```diff
@@ -0,0 +1,133 @@
+//! command definitions
+use error::Error;
+use error::ErrorKind;
+use fibers::{Executor, Spawn, ThreadPoolExecutor};
+use fibers_rpc::client::ClientServiceBuilder as RpcServiceBuilder;
+use futures::Future;
+use libfrugalos::client::frugalos::Client;
+use libfrugalos::entity::bucket::BucketId;
+use libfrugalos::entity::device::DeviceId;
+use libfrugalos::entity::object::ObjectId;
+use slog::Logger;
+use sloggers::types::Severity;
+use std::collections::BTreeSet;
+use std::net::SocketAddr;
+use std::path::PathBuf;
+use structopt::StructOpt;
+use trackable::error::ErrorKindExt;
+use Result;
+
+pub mod object;
+
+#[allow(missing_docs)]
+pub struct OneshotCommandContext {
+    pub logger: Logger,
+    pub executor: ThreadPoolExecutor,
+    pub frugalos_client: Client,
+}
+
+impl OneshotCommandContext {
+    #[allow(missing_docs)]
+    pub fn new(logger: Logger, rpc_addr: SocketAddr) -> Result<Self> {
+        let executor = track!(ThreadPoolExecutor::with_thread_count(1).map_err(Error::from))?;
```

Comment:
FYI. [fibers_global](https://docs.rs/fibers_global/0.1.0/fibers_global/) may be useful for command line tools (but the current version does not have an option to change the size of thread pool...).